crates/bpf-features-macros/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+use bpf_features::BpfFeatures;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, LitStr};
+
+/// Generates a mapping of `BpfFeatures` to eBPF program bytes. The bytes are
+/// embedded into the final binary thanks to `include_bytes_aligned`.
+#[proc_macro]
+pub fn ebpf_programs_map(input: TokenStream) -> TokenStream {
+    let probe = parse_macro_input!(input as LitStr).value();
+
+    let mut tokens = quote! {};
+
+    for (features, (bpf_objfile_suffix, _)) in BpfFeatures::all_combinations() {
+        let path = format!(
+            "{}/{}.{}",
+            std::env::var("OUT_DIR").unwrap(),
+            probe,
+            bpf_objfile_suffix
+        );
+        tokens = quote! {
+            #tokens
+            (#features, bpf_common::aya::include_bytes_aligned!(#path)),
+        };
+    }
+
+    let tokens = quote! {
+        [#tokens]
+    };
+
+    tokens.into()
+}
```

Comment:
This proc macro is not needed, you are only parsing the whole input as a string.

It's possible to take this code and put it directly in the other declarative macro `ebpf_program`.

Reply:
Unfortunately it's not possible - `include_bytes_aligned!` requires string literals at compile time. If you try to process `BpfFeatures::all_combinations()` directly in `ebpf_program` declarative macro, you will end up with strings known at runtime instead of literals known at compile time.

Reply:
ok